internal/token/token_test.go

```diff
@@ -0,0 +1,143 @@
+package token_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd-oidc-brokers/internal/fileutils"
+	"github.com/ubuntu/authd-oidc-brokers/internal/providers/info"
+	"github.com/ubuntu/authd-oidc-brokers/internal/token"
+	"golang.org/x/oauth2"
+)
+
+var testToken = token.AuthCachedInfo{
+	Token: &oauth2.Token{
+		AccessToken:  "accesstoken",
+		RefreshToken: "refreshtoken",
+	},
+	RawIDToken: "rawidtoken",
+	UserInfo: info.User{
+		Name:  "foo",
+		UUID:  "saved-user-id",
+		Home:  "/home/foo",
+		Gecos: "foo",
+		Shell: "/usr/bin/bash",
+		Groups: []info.Group{
+			{Name: "token-test-group", UGID: "12345"},
+		},
+	},
+}
+
+func TestCacheAuthInfo(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		setup       func(t *testing.T, path string)
+		expectError bool
+	}{
+		"Successfully store token with non-existing parent directory": {
+			setup: func(t *testing.T, path string) {
+				// No setup needed
+			},
+		},
+		"Successfully store token with existing parent directory": {
+			setup: func(t *testing.T, path string) {
+				err := os.MkdirAll(filepath.Dir(path), 0700)
+				require.NoError(t, err, "MkdirAll should not return an error")
+			},
+		},
+		"Successfully store token with existing file": {
+			setup: func(t *testing.T, path string) {
+				err := os.MkdirAll(filepath.Dir(path), 0700)
+				require.NoError(t, err, "MkdirAll should not return an error")
+				err = token.CacheAuthInfo(path, testToken)
+				require.NoError(t, err, "CacheAuthInfo should not return an error")
```

Comment:
We still have the setup functions in the test cases...

Reply:
I changed that now

Reply:
Well, you fixed that on one of the tests... But the other still have them

Reply:
Oops. I think I updated them all now.